craft_providers/lxd/launcher.py

```diff
@@ -30,72 +30,74 @@
-        image.
-
-    :returns: Name of (compatible) snapshot to use.
-    """
-    return "-".join(
-        [
-            "snapshot",
-            image_remote,
-            image_name,
-            compatibility_tag,
-        ]
-    )
-
-
-def _publish_snapshot(
+def _create_instance(
     *,
-    lxc: LXC,
-    snapshot_name: str,
     instance: LXDInstance,
+    base_instance: Optional[LXDInstance],
     base_configuration: Base,
+    image_name: str,
+    image_remote: str,
+    ephemeral: bool,
+    map_user_uid: bool,
+    uid: Optional[int],
+    project: str,
+    remote: str,
+    lxc: LXC,
 ) -> None:
-    """Publish snapshot from instance.
+    """Launch and setup an instance from an image.
+
+    If a base instance is passed, copy the instance to the base instance.
 
-    Stop instance and publish its contents to an image with the specified alias.
-    Once published, restart instance and ensure it is ready for use.
+    Preconditions: The LXD project exists and the instance and base instance (if
+    defined) do not exist.
 
+    :param instance: LXD instance to launch and setup
+    :param base_instance: LXD instance to be created as a copy of the instance. If the
+    base_instance is None, a base instance will not be created.
+    :param base_configuration: Base configuration to apply to instance.
+    :param image_name: LXD image to use, e.g. "20.04".
+    :param image_remote: LXD image to use, e.g. "ubuntu".
+    :param ephemeral: After the instance is stopped, delete it.
+    :param map_user_uid: Map host uid/gid to instance's root uid/gid.
+    :param uid: The uid to be mapped, if ``map_user_id`` is enabled.
+    :param project: LXD project to create instance in.
+    :param remote: LXD remote to create instance on.
     :param lxc: LXC client.
-    :param snapshot_name: Alias to use for snapshot.
-    :param instance: LXD instance to snapshot from.
-    :param base_configuration: Base configuration for instance.
     """
-    instance.stop()
-
-    lxc.publish(
-        alias=snapshot_name,
-        instance_name=instance.instance_name,
-        force=True,
-        project=instance.project,
-        remote=instance.remote,
+    logger.warning(
+        "Creating new instance from image %r from remote %r.", image_name, image_remote
     )
+    instance.launch(
+        image=image_name,
+        image_remote=image_remote,
+        ephemeral=ephemeral,
+        map_user_uid=map_user_uid,
+        uid=uid,
+    )
+    base_configuration.setup(executor=instance)
 
-    # Restart container and ensure it is ready.
-    instance.start()
-    base_configuration.wait_until_ready(executor=instance)
+    if base_instance is not None:
+        logger.warning(
+            "Creating new base instance %r from instance.", base_instance.instance_name
+        )
+        # stop the instance before copying, to ensure it is in a "good" state
+        instance.stop()
```

Comment:
The output of the prior `setup()` call means that all the system calls (`apt`, `snapd`, `systemctl`) completed successfully but the application hasn't been able to interact with the instance yet.  So the instance is "idling".

And yes, the later `start()` is "resuming" the instance.

Side note - you can copy a running instance, but LXD warns that it may cause data corruption.